### PR TITLE
Add Exporter.Reader and remove metric.Reader embedding

### PR DESCRIPTION
**Pull request by pellared**
Fixes https://github.com/open-telemetry/opentelemetry-go/issues/4310

CC @dashpole 

I tend to use type embedding only one I see no other option. E.g.  https://pkg.go.dev/go.opentelemetry.io/otel/metric/embedded is an example when type embedding is handy and there is no other option.

I feel that it is more readable to have 

```go
metric.NewMeterProvider(metric.WithReader(exporter.Reader()))
```

instead of

```go
metric.NewMeterProvider(metric.WithReader(exporter))
```

as without reading the docs one will not know that exporter implements `metric.Reader`.

EDIT:

Moreover, this API is safer as that the user cannot modify the `exporter.Reader` after construction using `prometheus.New` which could lead to race conditions.

### CODE REVIEW

File: exporters/prometheus/exporter.go [hunk 1]
```diff
@@ -47,13 +47,16 @@ const (
...
+// Exporter is a Prometheus Exporter that registers a prometheus.Collector
+// and encapsulates metric.Reader used for integration with a MeterProvider.
 type Exporter struct {
-	metric.Reader
+	reader metric.Reader
 }
 
-var _ metric.Reader = &Exporter{}
+// Reader returns the exporter's metrics reader.
+func (e *Exporter) Reader() metric.Reader {
```
Comment (dashpole) [inline exporters/prometheus/exporter.go:57]: Does this still allow the Reader to be accessed and modified? (from https://github.com/open-telemetry/opentelemetry-go/issues/4310)

**Comment by dashpole** [comment]
Curious for others' thoughts on readability.

File: exporters/prometheus/exporter.go (hunk 1 above)
Comment (pellared) [reply exporters/prometheus/exporter.go:57]: > Does this still allow the Reader to be accessed 

Yes. Via `reader.Reader()`. I was concerned mostly about possible modification after construction which would then lead to race conditions.

> modified

The whole instance cannot be "swapped". Meaning that the `exporter.reader` cannot be set to another instance after the construction.

Comment (dashpole) [reply exporters/prometheus/exporter.go:57]: That makes sense, thanks.

Comment (pellared) [reply exporters/prometheus/exporter.go:57]: I tried to add it to the PR description.

**Comment by dmathieu** [comment]
I agree with @MrAlias that the current way is more readable.
Especially since there shouldn't be anything to do with an exporter but get its reader.

**Pull request closed without merging**